### PR TITLE
core: Add Redis policy store

### PR DESCRIPTION
**Pull request by 115100**
This is a requirement of https://github.com/ory-am/hydra/issues/306

```bash
$ go test
[...]
PASS
ok      github.com/ory-am/ladon 29.019s
```

### CODE REVIEW

File: manager_redis.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	payload, err := json.Marshal(policy)
+	if err != nil {
+		return err
+	}
+
+	wasKeySet, err := m.db.HSetNX(m.redisPoliciesKey(), policy.GetID(), string(payload)).Result()
+	if !wasKeySet {
+		return errors.New("Policy exists")
+	} else if err != nil {
+		return err
```
Comment (arekkas) [inline manager_redis.go]: Please wrap errors in `errors.Wrap(err, "Description: ")`

File: manager_redis.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+// Delete removes a policy.
+func (m *RedisManager) Delete(id string) error {
+	return m.db.HDel(m.redisPoliciesKey(), id).Err()
+}
+
+// FindPoliciesForSubject finds all policies associated with the subject.
+func (m *RedisManager) FindPoliciesForSubject(subject string) (Policies, error) {
+	var ps Policies
+
+	iter := m.db.HScan(m.redisPoliciesKey(), 0, "", 0).Iterator()
```
Comment (arekkas) [inline manager_redis.go]: Are you simply fetching all policies to memory and then iterating on them here?

Comment (115100) [inline manager_redis.go]: By default, HScan will return 10 results at a time according to http://redis.io/commands/scan.

**Comment by 115100** [comment]
Wrapped errors. Let me know if you're happy with that and I'll squash.

**Comment by arekkas** [comment]
Looks good! I will test this code tomorrow for some final reviewing and going to merge it then. No squashing required!

One more thing, please sign your commit with `git commit --amend -s` :)

**Comment by arekkas** [comment]
thanks!

**Pull request merged**